tcms/templates/management/get_tag.html

```diff
@@ -16,19 +16,11 @@
 				<td><a href="{% url "plans-all" %}?action=search&tag__name__in={{ tag }}" title="{{ tag.num_plans }} plans tagged {{ tag }}">{{ tag.num_plans }}</a></td>
 				<td><a href="{% url "testcases-search" %}?a=search&tag__name__in={{ tag }}"  title="{{ tag.num_cases }} cases tagged {{ tag }}">{{ tag.num_cases }}</a></td>
 				<td><a href="{% url "testruns-all" %}?action=search&tag__name__in={{ tag }}"  title="{{ tag.num_runs }} runs tagged {{ tag }}">{{ tag.num_runs }}</a></td>
-				{% if object.plan_id and perms.testplans.delete_testplantag %}
-				<td><a href="#tag" class="remove js-remove-tag" title="remmove tag from this plan" >Remove</a>
-				<a href="#tag" class="edit js-edit-tag" title="edit tag" >Edit</a>
-				</td>
-				{% else %}
-				{% if object.case_id and perms.testcases.delete_testcasetag %}
-				<td><a href="#tag" class="remove js-remove-tag" title="remmove tag from this case" >Remove</a>
-				<a href="#tag" class="edit js-edit-tag" title="edit tag" >Edit</a>
-				</td>
+				{% if object.plan_id and perms.testplans.delete_testplantag or object.case_id and perms.testcases.delete_testcasetag %}
```

Comment:
You're missing the object.case_id condition but I'm wondering why do we check these conditions at all? Doesn't seem like object.plan_id or object.case_id is used inside this if block. Can you investigate if these IDs are used in the remove function or someplace else related to the remove functionality?

If the IDs conditions turn out not to be necessary then the if statement can check only on the permissions required.

Reply:
After investigating I found out the following: 

The plan/case id is indeed used when sending the delete request to the server. This request is sent by `tcms_actions.js `, where the following call is made 
`  var parameters = Nitrate.Utils.formSerialize(jQ('#id_tag_form')[0]);` 
The `parameters` object contains the params that are sent in the request, which looks like this `/management/tags/?a=remove&plan=1&tags=tagone` Therefore the `plan_id/case_id` is used and if it is not present in the request that would result in a server side error. However, I am not sure whether this outcome is possible in our situations, since the id comes from the form that visualises the tags for the given case/plan. Yet, I think that we may leave this check just to be sure that everything will be ok.